src/utils/validation.ts

```diff
@@ -31,3 +35,48 @@ export const isTestnet = (): boolean => {
 
   return network === 'mainnet' ? false : true;
 };
+
+export const getOnchainMetadataVersion = (onchainMetadata: Asset['onchain_metadata']): 1 | 2 => {
+  if (!onchainMetadata?.version) {
+    return 1;
+  }
+
+  if (Number(onchainMetadata.version) === 2) {
+    return 2;
+  }
+
+  return 1;
+};
+
+export const getOnchainMetadata = (
+  onchainMetadata: Asset['onchain_metadata'],
+  assetName: Asset['asset_name'],
+  policyId: Asset['policy_id'],
+) => {
+  if (!onchainMetadata || !assetName) return null;
+
+  const version = getOnchainMetadataVersion(onchainMetadata);
+
+  if (version === 1) {
+    assetName = Buffer.from(assetName, 'hex').toString('utf8');
```

Comment:
nitpick, mutating function parameters is not very nice!

Reply:
👍